### DISCUSSION TITLE
Does it support custom tile scheme?

### DISCUSSION THREAD

**Discussion by King-JXN** [maplibre/martin#1293] [Q&A]
I have a scene like this: Given a tile scheme, including resolution and scale, how does it create vector tile with target level in the tile scheme

**Comment by sharkAndshark** [comment]
No With [Table Source](https://maplibre.org/martin/sources-pg-tables.html) and Yes With [Function Source](https://maplibre.org/martin/sources-pg-functions.html)

## Not Support With Table Source

The answer would be not If you are using the [Table Source](https://maplibre.org/martin/sources-pg-tables.html) . As martin fetch tiles with this:

https://github.com/maplibre/martin/blob/59040a8686fbaa0390d6f0e8903774dab75fc320/martin/src/pg/query_tables.rs#L187-L210

See more on the postgis doc.
* [ST_TileEnvelope](https://postgis.net/docs/ST_TileEnvelope.html)
* [ST_AsMVTGeom](https://postgis.net/docs/ST_AsMVTGeom.html)

## Yes With Function Source

But with  [Function Source](https://maplibre.org/martin/sources-pg-functions.html) you could make tile envelop with your custom tile schema. So the answer would be yes.

**Comment by sharkAndshark** [comment]
More discussions on [issue 343](https://github.com/maplibre/martin/issues/343)
